src/shims/windows/fs.rs

```diff
@@ -419,6 +417,35 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     interp_ok(this.eval_windows("c", "FALSE"))
                 }
             }
+        } else if class == this.eval_windows_u32("c", "FileAllocationInfo") {
+            // File allocation size is independent of file EOF position on Windows, except that
+            // EOF must be <= allocation size. So we only do something if the file is bigger than
+            // the requested allocation size.
```

Comment:
So in other words, we don't grow, we only shrink? I don't understand what that has to do with "EOF position" vs "allocation size". I also don't know what those terms even mean. ;)

Reply:
EOF position is the file's length. Allocation size is generally a multiple of the underlying storage's sector size, and is effectively the file's pre-allocated capacity before writing will have to map more physical storage. 

Rust `std` doesn't expose this cross-platform, and we don't support it in GetFileInformationByHandle, so the only _observable_ side effect of setting it is truncating length to be less than capacity.